src/components/AccountDrawer/DefaultMenu.tsx

```diff
@@ -17,14 +17,23 @@ enum MenuState {
   SETTINGS,
 }
 
-function DefaultMenu() {
+function DefaultMenu({ drawerOpen }: { drawerOpen: boolean }) {
   const { account } = useWeb3React()
   const isAuthenticated = !!account
 
   const [menu, setMenu] = useState<MenuState>(MenuState.DEFAULT)
   const openSettings = useCallback(() => setMenu(MenuState.SETTINGS), [])
   const closeSettings = useCallback(() => setMenu(MenuState.DEFAULT), [])
 
+  useEffect(() => {
+    if (!drawerOpen && menu === MenuState.SETTINGS) {
+      // wait for the drawer to close before resetting the menu
+      setTimeout(() => {
+        closeSettings()
```

Comment:
I think needs a cleanup effect to clear timeout

Reply:
I think I did it properly?

Reply:
weird, the diff shows me this:

<img width="844" alt="CleanShot 2023-08-02 at 13 03 00@2x" src="https://github.com/Uniswap/interface/assets/12100/d6e5d606-cbe9-4a14-b5c5-4d46b3330f19">

i went to edit it and then it showed me this:

<img width="584" alt="CleanShot 2023-08-02 at 13 02 56@2x" src="https://github.com/Uniswap/interface/assets/12100/7efe9a2a-5e03-415e-87d7-3e7c88b88340">


🤷